test/unit/UnitTestWallet.ts

```diff
@@ -122,9 +138,73 @@ describe('Wallet Unit Tests', () => {
       assert.isFalse(wallet.isWalletAbleToSendTx(0));
     });
 
-    it('errors if wallet not within range', async () => {
+    it('errors if wallet not within range', () => {
       wallet.create(1);
       expect(() => wallet.isWalletAbleToSendTx(1)).to.throw();
     });
   });
+
+  describe('signTransaction()', () => {
+    it('signs a transaction', async () => {
+      wallet.create(1);
+      const from = wallet.getAddresses()[0];
+      const nonce = config.web3.toHex(await wallet.getNonce(from));
+
+      const sig = await wallet.signTransaction(from, nonce, opts);
+      expect(sig).to.haveOwnProperty('r');
+      expect(sig).to.haveOwnProperty('s');
+      expect(sig).to.haveOwnProperty('v');
+    });
+  });
+
+  describe('sendFromIndex()', () => {
+    it('returns ignore when not enough balance on account', async () => {
+      wallet.create(1);
+
+      const receipt = await wallet.sendFromIndex(0, opts);
+      assert.equal(receipt.ignore, true);
+    });
+
+    it('returns ignore when sendint a Tx is in progress', async () => {
+      wallet.create(1);
+      const idx = 0;
+      const address = wallet.getAddresses()[idx];
+      wallet.walletStates[address] = {
+        sendingTxInProgress: true
+      };
+
+      const receipt = await wallet.sendFromIndex(idx, opts);
+      assert.equal(receipt.ignore, true);
+    });
+
+    it('returns receipt when is able to send the transaction', async () => {
+      wallet.create(1);
+      const idx = 0;
+      const address = wallet.getAddresses()[idx];
+
+      const txHash = await Bb.fromCallback((callback: any) =>
+        config.web3.eth.sendTransaction(
+          {
+            from: myAccount,
+            to: address,
+            value: config.web3.toWei('0.5', 'ether')
+          },
+          callback
+        )
+      );
+      assert.equal(txHash.length, 66);
+
+      const receipt = await wallet.sendFromIndex(idx, opts);
+      expect(receipt).to.haveOwnProperty('from');
+      expect(receipt).to.haveOwnProperty('receipt');
+    });
+  });
+
+  describe('sendFromNext()', () => {
+    it('returns generates a random index and sends from it', async () => {
```

Comment:
The index isn't random it goes sequentially

Reply:
True, misunderstood that. Corrected.